### PR TITLE
ci: add step to run go mod tidy across the code bases

### PR DESCRIPTION
**Pull request by burmudar**
Fixes https://github.com/sourcegraph/devx-support/issues/355

#### Example run
Screenshot 2023-10-31 at 13 11 51

## Test plan
CI

### CODE REVIEW

File: dev/ci/bazel-gomodtidy.sh [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  echo "--- :bazel: Running go mod tidy in $dir"
+  bazel run @go_sdk//:bin/go -- mod tidy
+  cd -
+}
+
+# search for go.mod and run `go mod tidy` in the directory containing the go.mod
+find . -name go.mod -type f -exec dirname '{}' \; | while read -r dir; do runGoModTidy ${dir}; done
+# check if go.mod got updated
+git ls-files --exclude-standard --others | grep go.mod | xargs git add --intent-to-add
+diffFile=$(mktemp)
```
Comment (varungandhi-src) [inline dev/ci/bazel-gomodtidy.sh:19]: Will this temp file be automatically cleaned up or can these accumulate?

**Review by varungandhi-src — APPROVED** [review]
Thanks!

File: dev/ci/bazel-gomodtidy.sh (hunk 1 above)
Comment (burmudar) [reply dev/ci/bazel-gomodtidy.sh:19]: I expected that the os would clean it up, but reading up that isn't the case so I'll add a trap to cleanup on exit

File: dev/ci/bazel-gomodtidy.sh [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+# check if go.mod got updated
+git ls-files --exclude-standard --others | grep go.mod | xargs git add --intent-to-add
+
+diffFile=$(mktemp)
+trap 'rm -f "${diffFile}"' EXIT
+
+git diff --color=always --output="${diffFile}" --exit-code || EXIT_CODE=$?
+
+# if we have a diff, go.mod got updated so we notify people
+if [[ $EXIT_CODE -ne 0 ]]; then
```
Comment (jhchabran) [inline dev/ci/bazel-gomodtidy.sh:26]: It took me a second to understand why this works, then I realized I didn't understand, then I tested things and discovered that the var being undefined means it is equal to zero. 

I suppose this is to get around the `set -e` right?

**Review by jhchabran — APPROVED** [review]

**Pull request merged**